hledger-lib/Hledger/Data/Transaction.hs

```diff
@@ -228,6 +228,7 @@ postingAsLines elideamount onelineamounts ps p = concat [
     shownAmounts
       | elideamount    = [""]
       | onelineamounts = [fitString (Just amtwidth) Nothing False False $ showMixedAmountOneLine $ pamount p]
+      | null (amounts $ pamount p) = [""]
```

Comment:
I had to change this because otherwise postings with empty amount would not be printed, even though they should be in the case of a balance-only transaction. I don't really see why a posting with no amounts (but potentially an account/balance/...) should not be printed in general actually, so I figured this change made sense.